packages/features/bookings/Booker/components/hooks/useSlots.ts

```diff
@@ -129,9 +130,9 @@ export const useSlots = (event: { data?: Pick<BookerEvent, "id" | "length"> | nu
 
   // In case of skipConfirm flow selectedTimeslot would never be set and instead we could have multiple tentatively selected timeslots, so we pick the latest one from it.
   const timeSlotToBeBooked = selectedTimeslot ?? allSelectedTimeslots.at(-1);
-
+  const isReservationSupported = isCookieCreationAllowed();
   const handleReserveSlot = () => {
-    if (eventTypeId && timeSlotToBeBooked && eventDuration) {
+    if (eventTypeId && timeSlotToBeBooked && eventDuration && isReservationSupported) {
```

Comment:
We shouldn't reserve a slot if we can't identify the client in the next request. It avoids a bug where a person could endup reserving any number of slots, just by browsing through the slots.